### PR TITLE
Adds Example of using State in Effects

### PR DESCRIPTION
**Pull request by austbot**
Referencing #560

### CODE REVIEW

File: docs/effects/README.md [hunk 1]
```diff
@@ -99,8 +99,26 @@ import { AdminEffects } from './effects/admin';
...
-## API Documentation
+```ts
+...
+import 'rxjs/add/operator/withLatestFrom';
+...
+class SomeEffects {
+@Effect() someEffect$ = this.actions$
+    .ofType(SomeActions.SOME_ACTION)
+    .withLatestFrom(this.state$) # merging state in
+    .map(([action, state]) => state.search.query)
```
Comment (felikf) [inline docs/effects/README.md]: Nice to have: add types:

```
.map(([action, state]: ([SomeActionType, SomeStateSliceType])) => state.search.query)
```

File: docs/effects/README.md [hunk 2]
```diff
@@ -99,8 +99,26 @@ import { AdminEffects } from './effects/admin';
 })
 export class AdminModule {}
 ```
+## Effects that use State
```
Comment (brandonroberts) [inline docs/effects/README.md]: This should go as a section in the API. Generally, you should avoid selecting state in your effects, but instead provide any data the effect needs as part of the action payload.

**Comment by michaelchiche** [comment]
shouldn't the example use the "new"  rxjs 5.5 syntax with lettable operators and use the pipe function?

**Comment by brandonroberts** [comment]
@michaelchiche yes. That will be a part of a bigger sweep of the docs to update all the examples.

**Comment by brandonroberts** [comment]
@austbot are you still planning on landing this change?

**Comment by brandonroberts** [comment]
Closing for now. Re-open if you still want to land this.

**Pull request closed without merging**